lib/mix/lib/mix/cli.ex

```diff
@@ -76,6 +78,8 @@ defmodule Mix.CLI do
     try do
       ensure_no_slashes(name)
       Mix.Task.run("loadconfig")
+      # Restart Logger so it uses the configuration of the project.
+      if logger_configured?(), do: restart_logger()
```

Comment:
I thought it would be nice to only restart the logger if it's configured by the application :)

Reply:
I like it.

Reply:
Unfortunately, I think this won't work if the logger is configured using the `:env` key in `mix.exs` :(

Reply:
Oh! Nevermind. `:env` is scoped per app 😬